retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs

```diff
@@ -0,0 +1,357 @@
+        }
+        catch (Exception)
+        {
+            Console.WriteLine($"Table {fullTableId} does not exist.");
+        }
+
+        Console.WriteLine($"Creating BigQuery Table {fullTableId}");
+
+        TableSchemaBuilder tableSchemaBuilder = new TableSchemaBuilder();
+
+        // Parsing json schema.
+        string jsonSchema = File.ReadAllText(tableSchemaFilePath);
+
+        JsonTextReader jsonReader = new JsonTextReader(new StringReader(jsonSchema))
+        {
+            SupportMultipleContent = true
+        };
+
+        JsonSerializer jsonSerializer = new JsonSerializer();
+
+        while (jsonReader.Read())
+        {
+            TableFieldSchema tableFieldSchema = jsonSerializer.Deserialize<TableFieldSchema>(jsonReader);
+
+            tableSchemaBuilder.Add(tableFieldSchema);
+        }
+
+        TableSchema finalTableSchema = tableSchemaBuilder.Build();
+
+        // Creating a BigQuery table.
+        try
+        {
+            BigQueryTable createdBigQueryTable = bigQueryClient.CreateTable(tableReference, finalTableSchema);
+
+            Console.WriteLine($"Created BigQuery Table {createdBigQueryTable.FullyQualifiedId}");
+
+            try
+            {
+                // Uploading json data to BigQuery table.
+                Console.WriteLine($"Uploading data from json to BigQuery table {createdBigQueryTable.FullyQualifiedId}");
+
+                using (FileStream sourceStream = File.Open(tableDataFilePath, FileMode.Open))
+                {
+                    BigQueryJob bigQueryJob = bigQueryClient.UploadJson(tableReference, finalTableSchema, sourceStream);
+
+                    var result = bigQueryJob.PollUntilCompleted();
+                }
+
+                BigQueryTable createdTable = bigQueryClient.GetTable(tableReference);
+
+                Console.WriteLine($"Uploaded {createdTable.Resource.NumRows} rows to {createdTable.FullyQualifiedId}");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Table {createdBigQueryTable.FullyQualifiedId} was not populated with data. Error: {ex.Message}");
+            }
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"Table {tableName} was not created. Error: {ex.Message}");
+        }
+    }
+}
+
+/// <summary>
+/// Create test resources.
+/// </summary>
+
+public static class CreateTestResourcesTutorial
+{
+    private static readonly string ProductFileName = "products.json";
+    private static readonly string EventsFileName = "user_events.json";
+
+    private static readonly string ProductDataSet = "products";
+    private static readonly string ProductTable = "products";
+    private static readonly string EventsDataSet = "user_events";
+    private static readonly string EventsTable = "events";
+
+    private static readonly string productsBucketName = Environment.GetEnvironmentVariable("BUCKET_NAME");
```

Comment:
```suggestion
    private static readonly string productsBucketName = Environment.GetEnvironmentVariable("RETAIL_BUCKET_NAME");
```

Reply:
but in all tutorials just BUCKET_NAME without RETAIL

Reply:
We have some other samples on CI (not Retail) using the BUCKET_NAME environment variable so Retail cannot use it. This was agreed months ago with @t-karasova and `RETAIL_BUCKET_NAME` and `RETAIL_EVENTS_BUCKET_NAME` are the env variables we created on CI a while back, and everything has been working fine.

Do the tutorials texts refere to the env variables, etc.? When we agreed to this it wasn't a problem.

Reply:
yes, in all tutorials we have step when we create bucket and then export bucket name into this env variable: `export BUCKET_NAME=<YOUR_BUCKET_NAME>` and export BUCKET_NAME=<YOUR_BUCKET_NAME>

Reply:
I've sent an internal email about this.

Reply:
done